services/thermal_mgr/thermal_mgr.c

```diff
@@ -42,19 +43,66 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
-  /* Send an event to the thermal manager queue */
+  if (event == NULL) return ERR_CODE_INVALID_ARG;
+
+  if (xQueueSend(thermalMgrQueueHandle, event, 0) != pdPASS) {
+    return ERR_CODE_QUEUE_FULL;
+  }
+
+  return ERR_CODE_SUCCESS;
+}
+error_code_t thermalMgrSendEventToFront(thermal_mgr_event_t *event) {
+  if (event == NULL) return ERR_CODE_INVALID_ARG;
+
+  if (xQueueSendToFront(thermalMgrQueueHandle, event, 0) != pdPASS) {
```

Comment:
Need to check the queue handle isn't null, returning invalid state if this occurs